flake.nix

```
@@ -0,0 +1,61 @@
+{
+  nixConfig.allow-import-from-derivation = true;
+  inputs = {
+    nixpkgs.url = "github:nixos/nixpkgs/nixpkgs-unstable";
+    parts.url = "github:hercules-ci/flake-parts";
+    haskell-flake.url = "github:srid/haskell-flake";
+    pre-commit-hooks.url = "github:cachix/pre-commit-hooks.nix";
+    devshell.url = "github:numtide/devshell";
+    devshell.inputs.nixpkgs.follows = "nixpkgs";
+  };
+  outputs = inputs:
+    inputs.parts.lib.mkFlake {inherit inputs;} {
+      systems = ["x86_64-linux"];
+      imports = [
+        inputs.devshell.flakeModule
+        inputs.haskell-flake.flakeModule
+        inputs.pre-commit-hooks.flakeModule
+      ];
+
+      perSystem = {
+        config,
+        # pkgs,
+        # lib,
```

```suggestion
        pkgs,
```

This will be needed if you fix the `basePackages`, otherwise it can also be removed.